### PR TITLE
Adds a 1 second delay for table slamming and Dakimakuras

### PR DESCRIPTION
**Pull request by Poojawa**
## About The Pull Request

What it says on the tin

## Why It's Good For The Game

You slam your palms down on the table.x3515 can no longer happen within the span of a few ticks.

## Changelog
:cl:Poojawa
fix: fixed an oversight that allowed for massive client lagging.
/:cl:

### CODE REVIEW

**Comment by cebutris** [comment]
Just make tableslamming do minor damage if you do it more than once in a short timespan, easy.

**Comment by Poojawa** [comment]
I was told repeatedly to never implement an actual punishment for any actions. a click delay is superior to potential mis-clicks.

**Review by deathride58 — CHANGES_REQUESTED** [review]
This is way too long of a delay. Half of melee delay would be much more appropriate for harm intent, and 1 decisecond would be appropriate for tableslapping.

**Comment by Poojawa** [comment]
why does table slamming need to basically not have a delay, tho

**Comment by deathride58** [comment]
Half of melee delay is 4 deciseconds, which is actually quite a lot of delay (almost half a second!). Table slapping is different in that it's meant to be something you can spam, and a 1 decisecond delay lines up pretty well with how fast most people are able to click.

**Comment by Poojawa** [comment]
Except I've been hearing reports of people crashing others with click spam, so not having a delay is just... silly?

We already have massive amounts of to_chat lag with shotguns and the like.

**Comment by deathride58** [comment]
I'm not saying "Let there be no delay", i'm saying "Reduce the delay to something smaller, your proposed values are way too overkill for what you're trying to fix"
A 1 decisecond delay isn't going to cause massive amounts of clientside lag if people get together and spam it; clientside chat lag only occurs when there's a lot of lines to process in one single tick, which a 1 decisecond delay would be more than capable of preventing.
The reason why I propose separating the table slapping and table slamming delays is because table slamming is admittedly a little obnoxious for something that's supposed to be used primarily to get people's attention; a 4 decisecond click delay would be more than enough to curb the annoying table slamming spam without making the click delay on it overly obnoxious. Meanwhile, table slapping is meant to be spammable since, IRL, people who slap their hands on tables typically don't stop at just one slap, they typically drum their hands on the table, which a 1 decisecond click delay would be able to fully simulate; if a 1 decisecond click delay is truly terrible, then add a cooldown to the message itself.

File: code/game/objects/items/storage/dakis.dm [hunk 1]
```diff
@@ -22,7 +22,7 @@
 /obj/item/storage/daki/attack_self(mob/living/user)
 	var/body_choice
 	var/custom_name
-
+	user.setClickCooldown(1 SECONDS)
```
Comment (deathride58) [inline code/game/objects/items/storage/dakis.dm:25]: The hotkey for using items in your active hand bypasses click delay. This here won't affect the majority of dakispam instances.

**Pull request closed without merging**